Changing placement of lowered constant expressions

AlexeySotkin opened the pull request:
According to SPIRV specification (p2.4):
"All OpVariable instructions in a function must be in the first block in the
function. These instructions, together with any immediately preceding OpLine
instructions, must be the first instructions in that block."
Therefore, changing insertion point of lowered constant expressions from
begining of the function to the point preceding the current instruction.
This partially fixes #50

bader commented:
LGTM.

yxsamliu commented on lib/SPIRV/SPIRVLowerConstExpr.cpp:
```diff
@@ -132,18 +132,18 @@ SPIRVLowerConstExpr::visit(Module *M) {
           if (auto CE = dyn_cast<ConstantExpr>(Op)) {
             SPIRVDBG(dbgs() << "[lowerConstantExpressions] " << *CE;)
             auto ReplInst = CE->getAsInstruction();
-            ReplInst->insertBefore(FBegin->begin());
```
Can we insert it at the end of the first BB? Then redundant instructions are easier to be merged by optimizations.

AlexeySotkin replied:
Fixed

AlexeySotkin committed:
Changing placement of lowered constant expressions

According to SPIRV specification (p2.4):
"All OpVariable instructions in a function must be in the first block in the
function. These instructions, together with any immediately preceding OpLine
instructions, must be the first instructions in that block."
Therefore, lowered constant expressions must not be inserted at the beginig of
the basic block.
Insert it before the current instruction if the current basic block is the
first one in the function. Otherwise insert it at the end of the first basic
block in the function.
This partially fixes #50
